### DISCUSSION TITLE
Command mode

### DISCUSSION THREAD

**Discussion by CptPotato** [helix-editor/helix#5891] [Ideas]
I thought about whether or not it would be useful to have some kind of command mode or window where it's possible to enter multiple `:` commands with a history of previously executed commands and their results. This could also be a place to display log messages in-editor.

Ui-wise this is a rough mockup of what I had in mind:

![image](https://user-images.githubusercontent.com/3957610/217810312-31121ce4-8e1c-473d-aa05-88fb9fd80c7f.png)

Though I don't know if this would be of much help. Executing multiple commands in a row isn't very cumbersome and `:log-open` takes care of looking at the log already.

**Comment by LEI** [comment]
This is similar to the command-line window in [vim](https://vimhelp.org/cmdline.txt.html#q%3A) which is a modifiable buffer.

I don't think the logs should be there, though I coud see a use for a command which does not change the current file (equivalent to `:messages` in vim).

Another feature I miss is the partial history search behavior of readline with arrow up and down:
https://codeinthehole.com/tips/the-most-important-command-line-tip-incremental-history-searching-with-inputrc/
Maybe this was mentioned in another discussion?
